crates/ruff_db/src/vfs.rs

```diff
@@ -6,10 +6,47 @@ use dashmap::mapref::entry::Entry;
 pub use path::{VendoredPath, VendoredPathBuf, VfsPath};
 
 use crate::file_system::{FileRevision, FileSystemPath};
+use crate::vfs::private::FileStatus;
 use crate::{Db, FxDashMap};
 
 mod path;
 
+/// Interns a file system path and returns a salsa `File` ingredient.
+///
+/// Returns `None` if the path doesn't exist, isn't accessible, or if the path points to a directory.
+#[inline]
+pub fn file_system_path_to_file(db: &dyn Db, path: impl AsRef<FileSystemPath>) -> Option<VfsFile> {
```

Comment:
I would consider naming this function `system_path_to_file`. It's more concise, and IMO having `file` twice in the name is a little confusing.